code_generation/templates/power_grid_model_c/power_grid_model_c/include/power_grid_model_c/dataset_class_maps.h.jinja

```diff
@@ -0,0 +1,47 @@
+// SPDX-FileCopyrightText: 2022 Contributors to the Power Grid Model project <dynamic.grid.calculation@alliander.com>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+// This header file is automatically generated. DO NOT modify it manually!
+
+// clang-format off
+
+/**
+ * @brief header file which includes helper extern global variables of meta data pointers 
+ *    to all datasets, compoments, and attributes
+ *
+ */
+
+#pragma once
+#ifndef POWER_GRID_MODEL_C_DATASET_DEFINITIONS_H
+#define POWER_GRID_MODEL_C_DATASET_DEFINITIONS_H
+
+#include "basics.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+{% for dataset, components in all_map.items() -%}
+// dataset {{ dataset }}
+PGM_API extern PGM_MetaDataset const* const PGM_def_{{ dataset }};
+// components of {{ dataset }}
+{% for component, attributes in components.items() -%}
+// component {{ component }}
+PGM_API extern PGM_MetaComponent const* const PGM_def_{{ dataset }}_{{ component }};
+// attributes of {{ dataset }} {{ component }}
+{% for attribute in attributes -%}
+PGM_API extern PGM_MetaAttribute const* const PGM_def_{{ dataset }}_{{ component }}_{{ attribute }};
+{% endfor -%}
+{% endfor -%}
+{% endfor -%}
```

Comment:
maybe a nitpick but should we add additional whitelines here? like

```suggestion
{% for dataset, components in all_map.items() -%}
// dataset {{ dataset }}
PGM_API extern PGM_MetaDataset const* const PGM_def_{{ dataset }};
// components of {{ dataset }}
{% for component, attributes in components.items() -%}
// component {{ component }}
PGM_API extern PGM_MetaComponent const* const PGM_def_{{ dataset }}_{{ component }};
// attributes of {{ dataset }} {{ component }}
{% for attribute in attributes -%}
PGM_API extern PGM_MetaAttribute const* const PGM_def_{{ dataset }}_{{ component }}_{{ attribute }};

{% endfor -%}
{% endfor -%}
{% endfor -%}
```

Reply:
If I look at the currently generated file. The white space is okay in my opinion.